src/checkpoint/checkpoint_txn.c

```diff
@@ -2040,6 +2040,13 @@ __checkpoint_lock_dirty_tree(
             is_drop = true;
     }
 
+    /*
+     * Force the checkpoint if we're in live restore mode. Live restore needs to checkpoint the file
```

Comment:
Perhaps:
```
When using a live restore file system force every checkpoint. Live restore writes metadata for each file being restored, but it doesn't necessarily dirty the btree, this is a consequence of background thread migration. By forcing checkpoints we ensure that all open btrees will have their metadata updated. 

There were two alternative solutions here, which we list for posterity:
- Periodically dirtying the btree by calling __wt_tree_modify_set from the background thread.
- Perform a system wide checkpoint on completion of migration for every file migrated.
```

Can you also elaborate on these two options in the jira ticket and why we went with our current solution? We'll eventually move this content to the technical design doc.

Reply:
https://github.com/wiredtiger/wiredtiger/pull/11725/commits/93b128178f634cfddfa51fd636a31e8c4e18ad7b